src/pages/costModelsDetails/components/sourcesToolbar.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Button,
+  InputGroup,
+  InputGroupText,
+  Pagination,
+  PaginationProps,
+  TextInput,
+} from '@patternfly/react-core';
+import { ButtonVariant } from '@patternfly/react-core';
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarFilter,
+  DataToolbarItem,
+} from '@patternfly/react-core/dist/esm/experimental';
+import { SearchIcon } from '@patternfly/react-icons';
+import React from 'react';
+import { InjectedTranslateProps } from 'react-i18next';
+import { Omit } from 'react-redux';
+
+interface SearchInputProps {
+  id: string;
+  value: string;
+  onChange: (value: string, event: React.FormEvent<HTMLInputElement>) => void;
+  onSearch: (evt: React.KeyboardEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+}
+
+const SearchInput: React.SFC<SearchInputProps> = ({
+  id,
+  placeholder = '',
+  value,
+  onChange,
+  onSearch,
+}) => {
+  return (
+    <InputGroup>
+      <TextInput
+        value={value}
+        placeholder={placeholder}
+        id={id}
+        onChange={onChange}
+        onKeyPress={(evt: React.KeyboardEvent<HTMLInputElement>) => {
+          if (evt.key !== 'Enter' || value === '') {
+            return;
+          }
+          onSearch(evt);
+        }}
+      />
+      <InputGroupText style={{ borderLeft: '0' }}>
+        <SearchIcon />
+      </InputGroupText>
+    </InputGroup>
+  );
+};
+
+interface SourcesToolbarProps extends InjectedTranslateProps {
+  actionButtonProps: {
+    onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+  };
+  paginationProps: PaginationProps;
+  searchInputProps: Omit<SearchInputProps, 'placeholder'>;
+  filter: {
+    onRemove: (category: string, chip: string) => void;
+    onClearAll: () => void;
+    query: { name?: string[] };
+  };
+}
+
+export const SourcesToolbar: React.SFC<SourcesToolbarProps> = ({
+  t,
+  searchInputProps,
+  paginationProps,
```

Comment:
Just a suggestion... Instead of providing component props, you could provide the pagination and search components. Then, the table can interact with the component's callbacks directly and no need to pass in `t`.